### PR TITLE
Issue #19989: Add rounded corners to recently tabs' icon

### PR DESCRIPTION
**Pull request by rocketsroger**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by rocketsroger** [comment]
device-2021-07-12-190050

**Comment by rocketsroger** [comment]
@topotropic please confirm "Jump back in" icon corner is acceptable.  Thanks,

File: app/src/main/res/values/dimens.xml [hunk 1]
```diff
@@ -227,6 +227,10 @@
     <dimen name="top_sites_card_radius">8dp</dimen>
     <dimen name="top_sites_card_stroke_width">1dp</dimen>
 
+    <!-- Recent tab  -->
```
Comment (gabrielluong) [inline app/src/main/res/values/dimens.xml:230]:
```suggestion
    <!-- Recent tab -->
```

File: app/src/main/res/values/dimens.xml [hunk 2]
```diff
@@ -227,6 +227,10 @@
...
+    <dimen name="recent_tabs_icon_elevation">1dp</dimen>
```
Comment (gabrielluong) [inline app/src/main/res/values/dimens.xml:231]: Did we need this elevation? This was `0dp` for the favicon in top sites. See `top_sites_favicon_elevation`.

**Review by gabrielluong — APPROVED** [review]

**Comment by gabrielluong** [comment]
@rocketsroger you might want to provide a light theme screenshot so that the 1dp elevation is visible.

File: app/src/main/res/values/dimens.xml (hunk 1 above)
Comment (rocketsroger) [reply app/src/main/res/values/dimens.xml:230]: Thanks, I'll update

File: app/src/main/res/values/dimens.xml (hunk 2 above)
Comment (rocketsroger) [reply app/src/main/res/values/dimens.xml:231]: I got this value from `recentBookmarkFaviconShape`.  You're correct that it should be 0 dp.  As discussed, I'll update `recentBookmarkFaviconShape` to 0 dp as well.

**Comment by rocketsroger** [comment]
> @rocketsroger you might want to provide a light theme screenshot so that the 1dp elevation is visible.

Sure, I'll update the screenshot with light theme to show 0dp of elevation.  Thanks,

**Comment by rocketsroger** [comment]
device-2021-07-13-124223

**Review by topotropic — APPROVED** [review]

**Pull request merged**